compressor/js.py

```diff
@@ -50,3 +50,20 @@ def output(self, *args, **kwargs):
                     ret.append(subnode.output(*args, **kwargs))
                 return '\n'.join(ret)
         return super(JsCompressor, self).output(*args, **kwargs)
+
+    def filter_input(self, forced=False):
+        """
+        Passes each hunk (file or code) to the 'input' methods
+        of the compressor filters.
+        """
+        content = []
+        for hunk in self.hunks(forced):
+            # If a file ends with a function call, say, console.log()
+            # but doesn't have a semicolon, and the next file starts with
+            # a (, the individual files are ok, but when combined you get an
+            # error like TypeError...
+            # Forcing a semicolon in between fixes it.
+            if settings.COMPRESS_ENABLED or forced:
```

Comment:
@karyon OK - changed to just add ";" before each file.  The if is needed so semicolons aren't added before <script> tags when compression is disabled.  I'm not 100% confident in that 2-part test - does that seem reasonable?